### DISCUSSION TITLE
Generate only when graphql server is available

### DISCUSSION THREAD

**Discussion by gmencz** [dotansimha/graphql-code-generator#4033] [General] [answered]
I'm trying to watch for changes and generate the types but I need to wait until the graphql server is up and running to generate the types otherwise the generation fails, issue is my ts-node-dev is also watching for changes so when I change my schema, the server restarts and graphql-codegen tries to generate the types but because the server has not yet restarted, it fails, any ideas?

**Comment by ardatan** [comment, marked as answer]
I think you point to the URL endpoint as the schema in your codegen.yml file. You can point to your code files instead so codegen can take directly from your code.

**Reply by gmencz** [reply]
Thanks! that worked, I pointed to the file where the schema was defined as you suggested, thanks!
